src/qt/bitcoinamountlabel.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <qt/bitcoinamountlabel.h>
+
+#include <QChar>
+#include <QLabel>
+#include <QString>
+#include <QStringList>
+#include <QWidget>
+
+#include <cassert>
+
+namespace {
+const QChar space{' '};
+const QChar point{'.'};
+const QChar mask{'#'};
+
+QString cloak(QString value_and_unit)
```

Comment:
Basing it on the real number is revealing too much info.

Reply:
> Basing it on the real number is revealing too much info.

I'm not sure if I understand you correctly. Mind rewording your concerns?

Reply:
Your cloaking reveals at least how many whole number digits are in the real amount, by measuring the number of spaces you add.

https://github.com/bitcoin/bitcoin/commit/98ce0b830e505ddede5d56951e4ce7ee1bf5a0c5

Reply:
> Your cloaking reveals at least how many whole number digits are in the real amount, by measuring the number of spaces you add.

Fixed in the latest push.

> [98ce0b8](https://github.com/bitcoin/bitcoin/commit/98ce0b830e505ddede5d56951e4ce7ee1bf5a0c5)

Monospace font style prevents the UI to reflow the columns on every toggle (https://github.com/bitcoin/bitcoin/pull/16432#discussion_r305807357).